chat-widget/src/components/livechatwidget/common/startChatErrorHandler.ts

```diff
@@ -0,0 +1,116 @@
+            if (ex.httpResponseStatusCode === 400) {
+                logWidgetLoadFailed(ex);
+            } else {
+                logWidgetLoadCompleteWithError(ex);
+            }
+        } else if (ex.message === ChatSDKErrorName.InvalidConversation ||
+            ex.message === ChatSDKErrorName.ClosedConversation) {
+            logWidgetLoadCompleteWithError(ex);
+
+            // Reset all internal states
+            callingStateCleanUp(dispatch);
+            endChatStateCleanUp(dispatch);
+            closeChatStateCleanUp(dispatch);
+            chatSDKStateCleanUp(chatSDK);
+            DataStoreManager.clientDataStore?.removeData(getWidgetCacheIdfromProps(props));
+
+            // Starts new chat
+            dispatch({ type: LiveChatWidgetActionType.SET_CONVERSATION_STATE, payload: ConversationState.Closed });
+            return;
+        } else {
+            logWidgetLoadFailed(ex);
+        }
+    } else if (ex.message === "Authentication was not successful" ||
+        ex.message === "Network Error") {
+        logWidgetLoadCompleteWithError(ex);
+    }
+
+    NotificationHandler.notifyError(NotificationScenarios.Connection, "Start Chat Failed: " + ex);
+    dispatch({ type: LiveChatWidgetActionType.SET_START_CHAT_FAILING, payload: true });
+    if (!props?.controlProps?.hideErrorUIPane) {
+        // Set app state to failing start chat if hideErrorUI is not turned on
+        TelemetryHelper.logLoadingEvent(LogLevel.INFO, {
+            Event: TelemetryEvent.ErrorUIPaneLoaded,
+            Description: "Error UI Pane Loaded"
+        });
+    }
+    // Show the loading pane in other cases for failure, this will help for both hideStartChatButton case
+    dispatch({ type: LiveChatWidgetActionType.SET_CONVERSATION_STATE, payload: ConversationState.Loading });
+};
+
+const logWidgetLoadFailed = (ex?: ChatSDKError) => {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const exDetails: any = {
+        Exception: `Widget load complete with error: ${ex}`
+    };
+    if (ex?.httpResponseStatusCode) {
+        exDetails.HttpResponseStatusCode = ex.httpResponseStatusCode;
+    }
+    
+    TelemetryHelper.logLoadingEvent(LogLevel.ERROR, {
+        Event: TelemetryEvent.WidgetLoadFailed,
+        ExceptionDetails: exDetails,
+        ElapsedTimeInMilliseconds: TelemetryTimers?.WidgetLoadTimer?.milliSecondsElapsed
+    });
+};
+
+export const logWidgetLoadComplete = (additionalMessage?: string) => {
+    let descriptionString = "Widget load complete";
+    if (additionalMessage) {
+        descriptionString += `. ${additionalMessage}`;
+    }
+
+    TelemetryHelper.logLoadingEvent(LogLevel.INFO, {
+        Event: TelemetryEvent.WidgetLoadComplete,
+        Description: descriptionString,
+        ElapsedTimeInMilliseconds: TelemetryTimers?.WidgetLoadTimer?.milliSecondsElapsed
+    });
+};
+
+const logWidgetLoadCompleteWithError = (ex: ChatSDKError) => {
```

Comment:
It seems like `ex` is not going to be `ChatSDKError` if the exception is the "Authentication was not successful." case?

Reply:
This is safe because ex is "any" type in this case and there's null check for `ex.httpResponseStatusCode` inside `logWidgetLoadCompleteWithError` and `logWidgetLoadComplete`